microprofile-metrics/src/test/java/org/jboss/eap/qe/microprofile/metrics/MetricsTest.java

```diff
@@ -0,0 +1,320 @@
+     * @tpPassCrit Correct data are provided for metrics
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @RunAsClient
+    public void metricRegistryInjection() {
+        given().baseUri(deploymentUrl.toString()).basePath("summary/all-registries").accept(ContentType.JSON).get().then()
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("$", hasKey("base"),
+                        "base", hasItem("cpu.availableProcessors"),
+                        "base", hasItem("memory.usedHeap"),
+                        "base", hasItem("memory.committedHeap"),
+                        "base", hasItem("memory.maxHeap"),
+                        "base", hasItem("jvm.uptime"),
+                        "base", hasItem("thread.count"),
+                        "base", hasItem("thread.daemon.count"),
+                        "base", hasItem("thread.max.count"),
+                        "base", hasItem("classloader.loadedClasses.count"),
+                        "base", hasItem("classloader.loadedClasses.total"),
+                        "base", hasItem("classloader.unloadedClasses.total"),
+
+                        "$", hasKey("vendor"),
+                        "vendor.size()", greaterThan(0),
+
+                        "$", hasKey("app"),
+                        "app", hasItem("hello-count"),
+                        "app", hasItem("hello-time"),
+                        "app", hasItem("hello-freq"),
+                        "app", hasItem("hello-invocations"));
+    }
+
+    /**
+     * @tpTestDetails Smoke scenario to verify correct behaviour
+     * @tpPassCrit Correct data are provided for metrics
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @RunAsClient
+    public void getAllRegisteredMetricsOfGivenType() {
+        given().baseUri(deploymentUrl.toString()).basePath("summary/app-registry").accept(ContentType.JSON).get().then()
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("$", hasKey("app-counters"),
+                        "app-counters", hasSize(1),
+                        "app-counters", hasItem("hello-count"),
+
+                        "$", hasKey("app-timers"),
+                        "app-timers", hasSize(1),
+                        "app-timers", hasItem("hello-time"),
+
+                        "$", hasKey("app-meters"),
+                        "app-meters", hasSize(1),
+                        "app-meters", hasItem("hello-freq"),
+
+                        "$", hasKey("app-gauges"),
+                        "app-gauges", hasSize(0),
+
+                        "$", hasKey("app-concurrent-gauges"),
+                        "app-concurrent-gauges", hasSize(1),
+                        "app-concurrent-gauges", hasItem("hello-invocations"));
+    }
+
+    static class PrometheusMetricValueMatcher extends BaseMatcher {
```

Comment:
What is the purpose of this class? Please add a JavaDoc or a comment.

Reply:
@istraka Do you plan to add any comment here?

Reply:
Fixing...